packages/main/src/index.ts

```diff
@@ -44,34 +45,65 @@ const createWindow = async () => {
   mainWindow.on('ready-to-show', () => {
     mainWindow?.show();
 
-    if (import.meta.env.MODE === 'development') {
+    if (isDevelopment) {
       mainWindow?.webContents.openDevTools();
     }
   });
 
-  /**
-   * External hyperlinks open in the default browser.
-   *
-   * @see https://stackoverflow.com/a/67409223
-   */
-   mainWindow.webContents.setWindowOpenHandler(({ url }) => {
-    shell.openExternal(url);
-    return { action: 'deny' };
-  });
-
   /**
    * URL for main window.
    * Vite dev server for development.
    * `file://../renderer/index.html` for production and test
    */
-  const pageUrl = import.meta.env.MODE === 'development' && import.meta.env.VITE_DEV_SERVER_URL !== undefined
+  const pageUrl = isDevelopment && import.meta.env.VITE_DEV_SERVER_URL !== undefined
     ? import.meta.env.VITE_DEV_SERVER_URL
     : new URL('../renderer/dist/index.html', 'file://' + __dirname).toString();
 
 
   await mainWindow.loadURL(pageUrl);
 };
 
+ app.on('web-contents-created', (_event, contents) => {
+
+  /**
+   * Empty since there's no reason to enable any navigation.
+   *
+   * @see https://www.electronjs.org/docs/latest/tutorial/security#13-disable-or-limit-navigation
+   */
+  contents.on('will-navigate', (event, url) => {
+    const trustedOrigins = new Set();
+    const trustedProtocols = new Set();
+    const { origin, protocol } = new URL(url);
+    if (!trustedOrigins.has(origin) || !trustedProtocols.has(protocol)){
+      console.warn('Blocked navigating to an unrecognized origin:', origin);
+      event.preventDefault();
+    }
+  });
```

Comment:
I *used* to have an additional clause to _not_ block `will-navigate` if `url == 'localhost' && isDevelopment` because it was blocking hot-reloading. However after restarting my computer and `git clean -xdf` that error no longer occurs.

Reply:
Ah, I realized I needed that check to allow navigation for live-reloading of `index.html`.